evals/elsuite/audio/eval.py

```diff
@@ -47,82 +51,64 @@ def __init__(
-                    "image_url": {
-                        "url": f"data:audio/x-wav;base64,{base_64_wav}",
-                    },
-                }
-            )
-            if parts[1]:
-                content.append({"type": "text", "text": parts[1]})
+        prompt = self.build_prompt(sample, self.text_only)
+        kwargs = self.get_completion_kwargs(sample)
+        sampled = self.do_completion(prompt, **kwargs)
+        return self.compute_metrics(sample, sampled)
+
+    def run(self, recorder: RecorderBase):
+        samples = self.load_dataset()
+        self._recorder = recorder
+        self.eval_all_samples(recorder, samples)
+        return self.compute_corpus_metrics()
+
+    def load_dataset(self):
+        ds = load_hf_dataset(self.dataset, evals.eval._MAX_SAMPLES).cast_column(
+            "audio", Audio(sampling_rate=DEFAULT_SAMPLE_RATE)
+        )
+        return [Sample(data=row) for row in ds]
+
+    def get_completion_kwargs(self, sample: Sample):
+        return {}
 
-        prompt = [
-            {"role": "system", "content": "You are a helpful assistant."},
-            {"role": "user", "content": content},
-        ]
+    def do_completion(self, prompt, **kwargs):
         try:
             result = self.completion_fn(
                 prompt=prompt,
                 temperature=self.temperature,
                 max_tokens=self.max_tokens,
+                **kwargs,
             )
             sampled = result.get_completions()[0]
         except Exception as e:
-            if isinstance(content, list):
-                for part in content:
-                    if part["type"] == "image_url":
-                        part["image_url"]["url"] = "<audio data>"
+            redact_audio_content(m["content"] for m in prompt)
             logging.info("Sampling failed!")
-            logging.info(sample)
             logging.info(f"Prompt: {prompt}")
             logging.info(f"Error: {str(e)}")
             sampled = "ERROR: " + str(e)
-        return self.compute_metrics(sample, sampled)
+        return sampled
 
-    def run(self, recorder: RecorderBase):
-        samples = self._get_dataset()
-        self.eval_all_samples(recorder, samples)
 
-    def _get_dataset(self) -> List[Sample]:
-        parsed = urlparse(self.dataset)
-        query = parse_qs(parsed.query)
-        query = {k: v[0] for k, v in query.items()}
-        dataset = load_dataset(parsed.netloc + parsed.path, **query, streaming=True).cast_column(
-            "audio", Audio(sampling_rate=DEFAULT_SAMPLE_RATE)
-        )
-        if evals.eval._MAX_SAMPLES is not None:
-            dataset = dataset.take(evals.eval._MAX_SAMPLES)
-        return [self.load_sample(sample) for sample in dataset]
+class MatchAudioTask(AudioTask):
+    def get_match_events(self):
+        return self.recorder.get_events("match")
 
+    def get_expected_values(self):
+        return list(map(lambda e: e.data["expected"], self.get_match_events()))
```

Comment:
nit: why not use list comprehension?
```suggestion
        return [e.data["expected"] for e in self.get_match_events()]
```

Reply:
I just copied this code from another eval without looking too closely, definitely prefer your version